internal/moduletest/graph/apply.go

```diff
@@ -115,6 +119,34 @@ func (n *NodeTestRun) testApply(ctx *EvalContext, variables terraform.InputValue
 	// actually updated by this change. We want to use the run that
 	// most recently updated the tracked state as the cleanup
 	// configuration.
+	if state := ctx.GetFileState(key); state.backend.instance != nil {
+		if state.backend.run.Name == run.Name {
```

Comment:
There is another path where we update the state (See https://github.com/hashicorp/terraform/blob/0f260082701ede98db6c704b01eb504ceaab9db3/internal/moduletest/graph/apply.go#L69-L71). Wouldn't we want to also store the state in that path? If that is the case, Perhaps this logic could live be consolidated somewhere

Reply:
Thanks for highlighting this- I've included a write to persistent state via the backend in that block in [55d10a9](https://github.com/hashicorp/terraform/pull/36646/commits/55d10a96faa1ad1dbc0023cacf1ea0e2ddc2c6a2), and I've added a test that shows state is written to when an error occurs.